### ISSUE TITLE
Enumerable of fakes with random length

### ISSUE THREAD

**Issue by harrybellamy** [bchavez/Bogus#126]
Is it possible to generate an enumerable of a random number of fakes from a faker for that class?

e.g. some thing similar to the following:

```
var faker = new Faker<Foo>();

int minNumberOfFakes = 1;
int maxNumberOfFakes = 10;
IEnumerable<Foo> fakes = faker.Generate(minNumberOfFakes, maxNumberOfFakes);
```
After calling this, `fakes` would  contain between 1 and 10 elements.

**Edit** - updated the `Generate` call to use the min and max values arranged for it.

**Comment by bchavez** [comment]
Hi @harrybellamy,

Thanks for your question. I can think of 2 ways to extend the `Faker<T>` API with `Generate(min,max)`.
* First, via a custom derived faker:
https://github.com/bchavez/Bogus/blob/f7cd11db9eede6cf2d9421ab14641cb249c22e9d/Source/Bogus.Tests/GitHubIssues/Issue126.cs#L18-L39
* Second, via extension methods:
https://github.com/bchavez/Bogus/blob/f7cd11db9eede6cf2d9421ab14641cb249c22e9d/Source/Bogus.Tests/GitHubIssues/Issue126.cs#L41-L63

Hope that helps.
Brian

:hourglass_flowing_sand: :mag: ***["But I still haven't found what I'm for..."](https://www.youtube.com/watch?v=wdCJRybAtso)***

**Comment by harrybellamy** [comment]
Thanks Brian, much appreciated.

**Comment by bchavez** [comment]
Hi Harry,

No problem. I added a new extension method `.GenerateBetween(min, max)` in the `Bogus.Extensions` namespace that will be available in the next release. I've seen this issue in the wild before so it's probably time to add support for `GenerateBetween(min, max)`. For what it's worth, I decided to add `GenerateBetween` as an extension method since it has the feeling of indeterministic behavior which was counter to some of the core API ideas involving deterministic `.Generate()` semantics.

For now, I hope the few workarounds I gave are sufficient.

I'm holding back a new release mostly because I'm putting the finishing touches on some new upgrades and adding support for extended data set **NuGet** packages like `Bogus.Healthcare`, `Bogus.Entertainment`, and `Bogus.Literature`.

Thanks,
Brian

:cloud: :sunny: **[Bassnectar - Chasing Heaven - Into the Sun](https://www.youtube.com/watch?v=a2BAia075lE)**
